pkg/preflight/checks.go

```diff
@@ -200,6 +202,19 @@ func (c MemoryCheck) Run() (string, error) {
 	return "", nil
 }
 
+func (c DiskCheck) Run() (msg string, err error) {
+	disks, err := util.GetUniqueDisks()
+	if err != nil {
+		msg = fmt.Sprintf("Unable to get number of disks: %s", err.Error())
+		return
+	}
+
+	if len(disks) < 1 {
+		msg = "No disk detected. Harvester requires at least one disk."
```

Comment:
here the return `err` is nil, and it is not added to log, for pxe mode installation, is there some useful information booked in the log?

it seems below code is better to move msg processing before err check
```
		for _, c := range checks {
			msg, err := c.Run()
			if err != nil {
				// Preflight checks that fail to run at all are
				// logged, rather than killing the installer
				logrus.Error(err)
				continue
			}
			if len(msg) > 0 {
				preflightWarnings = append(preflightWarnings, msg)
			}
```

Reply:
As mentioned in the existing code comments, I believe the rationale for checking err first is: the non-nil `err` return from `c.Run()` indicates the failure of checking itself. It further implies that `msg` is not reliable and thus it's not appending to `preflightWarnings`.

So, I prefer we keep it as is.


> here the return err is nil, and it is not added to log, for pxe mode installation, is there some useful information booked in the log?

For those preflight warnings, I think they will be logged here for pxe mode installation:
https://github.com/harvester/harvester-installer/blob/f1f053c6359cfce28ddd922d385edc2224553740/pkg/console/install_panels.go#L2565-L2584